sdk/identity/identity/src/msal/types.ts

```diff
@@ -15,6 +15,11 @@ export interface MsalToken {
   expiresOn: Date | null;
 }
 
+/**
+ * Represents a valid (i.e. complete) MSAL token.
+ */
+export type ValidMsalToken = { [P in keyof MsalToken]-?: NonNullable<MsalToken[P]> };
```

Comment:
`-?` on the left hand side is called a [mapping modifier](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers) and makes the property required.

On the right hand side, [NonNullable](https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype) removes `null` (and undefined, but not optionality) from the union. 

End result is that an assertion function that `asserts thing is ValidMsalToken` ties runtime validation to types helping remove the `!` operators